### PR TITLE
Configure MCS labels if selinux is enabled

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/rancher/rke/issues/1520

As we are editing `GetProcessConfig` and Rancher also uses this function, in Rancher we will pass an empty Host to the function.

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -515,6 +516,8 @@ func IsContainerUpgradable(ctx context.Context, dClient *client.Client, imageCfg
 }
 
 func sliceEqualsIgnoreOrder(left, right []string) bool {
+	logrus.Debugf("Comparing, left: %s", left)
```
Comment (galal-hussein) [inline docker/docker.go]: This will generate tons of debug messages and will not really be useful to see which lists are we comparing, can we omit this

Comment (alena1108) [reply docker/docker.go]: It would make sense to log in debug when the result of comparison is false, so it's easier to figure out the discrepancy.

**Comment by nheinemans** [comment]
Any chance this will make it into 2.3?

**Comment by alena1108** [comment]
@nheinemans that's a plan

File: docker/docker.go (hunk 1 above)
Comment (superseb) [reply docker/docker.go]: This was scoped to checking if it changed, if so, it will log the diff not both the values.

**Comment by superseb** [comment]
Updates on the PR:
- Labels are now scoped to host with selinux enabled and to containers that share the service-sidekick container
- Changed the `sliceEqualsIgnoreOrder` function to only show the diff when its not equal.

**Pull request merged**